packages/snaps-utils/src/jsx.ts

```diff
@@ -0,0 +1,33 @@
+import type { JSXElement } from '@metamask/snaps-sdk/jsx';
+import { hasProperty, isPlainObject } from '@metamask/utils';
+
+/**
+ * Walk a JSX tree and call a callback on each node.
+ *
+ * @param node - The JSX node to walk.
+ * @param callback - The callback to call on each node.
+ */
+export function walkJsx(
+  node: JSXElement,
+  callback: (node: JSXElement) => void,
+) {
+  callback(node);
+
+  if (
+    hasProperty(node, 'props') &&
+    isPlainObject(node.props) &&
+    hasProperty(node.props, 'children')
```

Comment:
Can't a prop also be a component or do we not want to support that in our implementation?

I seem to recall some of the SIP having components as props

Reply:
None of the components use components in props at the moment.

Reply:
Once we want to do that, how do we remember to add support in this file and others?